.github/workflows/ci.yaml

```diff
@@ -3,6 +3,10 @@ on:
   push:
     branches:
       - '*'
+  pull_request:
+    branches:
+      - 'main'
+
```

Comment:
Regarding the Trivy scan, it currently scans the master branch on docker! We change this to a filesystem scan so that it locally downloads the files and does a local file scan on another PR branch instead!

If you want, I can work on it and raise a separate PR

Reply:
Oh alright! That makes sense! I like the dev dependencies being skipped for now!
Everything in the PR Lgtm now!

Reply:
If you think there's value in doing it another way, by all means please do open an issue!  I'm not saying this way is 100% correct or the absolute best/perfect way of doing it, just explaining why it was configured that way.  I'm all about improving things!